core/RankingQuery.php

```diff
@@ -92,11 +92,12 @@ class RankingQuery
     /**
      * Constructor.
      *
-     * @param int|false $limit The result row limit. See {@link setLimit()}.
+     * @param int $limit The result row limit. See {@link setLimit()}.
      */
-    public function __construct($limit = false)
+    public function __construct($limit = 0)
```

Comment:
Did you check what this method was doing if `$limit = 0` has been passed before? Looking at the code, that might have been a valid purpose maybe. So using 0 as default makes it impossible to set the limit to `0` on purpose.

Reply:
@sgiehl revert it back.